tests/kernel/interrupt/src/interrupt_offload.c

```diff
@@ -54,23 +58,26 @@ static void entry_offload_job(struct k_work *work)
 			"the offload did not run immediately.");
 	}
 
+	offload_job_cnt++;
```

Comment:
This is needed because in the scenario that when the ISR offload job four times to "identical" k_work, actually it only can be treated as the same "in progress" one, unless the previously submitted k_work has been done. Add the offload_job_cnt here is in order to verify how many times the offload jobs have done, then the "end_sem" needs the counts to make sure all the k_works have done before we check it.

Reply:
By "same deal," I meant `atomic_int`. Marking it as resolved.

Reply:
Oh, sorry, I misunderstood here, thank you!